### ISSUE TITLE
Sometimes, trips do not record

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#566]
Reported by Ryan.

> yesterday morning, I took a bike ride from 6:00 to 8:00am and it did not record any activity

**Comment by shankari** [comment]
Stan also reported that trips do not record for him. In his case, this is because he turned location settings off and on. We generated several notifications but he did not apparently respond to them

```
88,1597427170.9520001,2020-08-14T10:46:10.952000-07:00,TripDiaryStateMachineService : checkSelfPermission returned 0
97,1597427171.086,2020-08-14T10:46:11.086000-07:00,"TripDiaryStateMachineService : in restartFSMIfStartState, currState = local.state.waiting_for_trip_start"
```

```
207,1597428820.7410002,2020-08-14T11:13:40.741000-07:00,TripDiaryStateMachineService : checkSelfPermission returned 0
216,1597428820.924,2020-08-14T11:13:40.924000-07:00,"NotificationHelper : Generating notify with id 87225377, message Error 6 in location settings and pending intent PendingIntent{cf7ed5e: android.os.BinderProxy@496e53f}"
```

````
264,1597437324.786,2020-08-14T13:35:24.786000-07:00,TripDiaryStateMachineService : checkSelfPermission returned 0
272,1597437324.994,2020-08-14T13:35:24.994000-07:00,"NotificationHelper : Generating notify with id 87225377, message Error 6 in location settings and pending intent PendingIntent{cae4034: android.os.BinderProxy@9f95e5d}"
```

```
874,1597963260.696,2020-08-20T15:41:00.696000-07:00,TripDiaryStateMachineService : checkSelfPermission returned 0
891,1597963260.927,2020-08-20T15:41:00.927000-07:00,"NotificationHelper : Generating notify with id 87225377, message Error 6 in location settings and pending intent PendingIntent{242e6d6: android.os.BinderProxy@855ad57}"
```

Users need to keep their location tracking on all the time otherwise the geofences are deleted.

**Comment by shankari** [comment]
wrt Ryan's original report, here's what I saw for the trip from the morning of the 20th

The geofence exit actually happened at 11:27 the previous night. 

```
59684,1597858075.2879999,2020-08-19T11:27:55.288000-06:00,"TripDiaryStateMachineService : handleAction(local.state.waiting_for_trip_start, local.transition.exited_geofence) completed, waiting for async operations to complete"
```

However, due to #567 we got locations only sporadically, so the trip lingered for a long time.

```
61629,1597904727.103,2020-08-20T00:25:27.103000-06:00,"LocationChangeIntentService : FINALLY! Got location update, intent is Intent {  }"
61653,1597911482.56,2020-08-20T02:18:02.560000-06:00,"LocationChangeIntentService : FINALLY! Got location update, intent is Intent { }"
```

Until around 5am when we started getting points quickly

```
61677,1597921524.893,2020-08-20T05:05:24.893000-06:00,"LocationChangeIntentService : FINALLY! Got location update, intent is Intent {  }"
61701,1597922130.7470002,2020-08-20T05:15:30.747000-06:00,"LocationChangeIntentService : FINALLY! Got location update, intent is Intent {  }"
61725,1597922495.457,2020-08-20T05:21:35.457000-06:00,"LocationChangeIntentService : FINALLY! Got location update, intent is Intent {  }"
```

and finally detected the trip end.

```
68723,1597931766.76,2020-08-20T07:56:06.760000-06:00,"TripDiaryStateMachineService : handleAction(local.state.ongoing_trip, local.transition.stopped_moving) called"
```

This could also explain #570 because the trip actually started on the 19th, so trip would show up on the diary for that day, and not the day that the trip actually happened.

**Comment by shankari** [comment]
Heard back from R again. It turns out that this is because the trip spans two days, which we currently don't support for unprocessed trips. So when we read the trips for the 19th, we skip it because it doesn't end, and when we read the trips for the 20th, we skip it because it doesn't start. So this is because of #567 

If we had sent this to the server, we would in fact be able to see the trip because we do handle cross-day trips in the more complex trip segmentation logic on the server. However, because of #568 we weren't uploading any trips.

So once we fix #567 and #568, this should be fixed as well.
